### PR TITLE
add nancumprod and nancumsum funcs

### PR DESCRIPTION
**Pull request by KsanaKozlova**

### CODE REVIEW

File: tests/test_mathematical.py [hunk 1]
```diff
@@ -24,3 +26,65 @@ def test_diff(array):
     result = inp.linalg.det(ia)
     expected = numpy.linalg.det(a)
     numpy.testing.assert_allclose(expected, result)
+
+
+class Test_nancumsum(unittest.TestCase):
```
Comment (densmirn) [inline tests/test_mathematical.py:31]:
```suggestion
class TestNanCumSum(unittest.TestCase):
```

File: tests/test_mathematical.py [hunk 2]
```diff
@@ -24,3 +26,65 @@ def test_diff(array):
...
+    def test_nancumsum_dim2_nan(self):
+        a = numpy.array([[1, 2, numpy.nan], [3, -4, -5]])
+        ia = dpnp.array(a)
+
+        result = dpnp.nancumsum(ia)
+        expected = numpy.nancumsum(a)
+        numpy.testing.assert_array_equal(expected, result)
+
+
+class Test_nancumprod(unittest.TestCase):
```
Comment (densmirn) [inline tests/test_mathematical.py:60]:
```suggestion
class TestNanCumProd(unittest.TestCase):
```

File: tests/test_mathematical.py [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 import numpy
 
+import unittest
```
Comment (densmirn) [inline tests/test_mathematical.py:7]: We run tests by `pytest`. So we are able to not use `unittests` like in `cupy` tests.

File: dpnp/backend_mathematical.pyx [hunk 4]
```diff
@@ -283,6 +285,66 @@ cpdef dparray dpnp_multiply(dparray x1, x2):
...
+        if dpnp.isnan(x1[i]):
+            cur_res *= 1
+        else:
+            cur_res *= x1[i]
+        result._setitem_scalar(i, cur_res)
+
+    return result
+
+
+cpdef dparray dpnp_nancumsum(dparray x1):
```
Comment (densmirn) [inline dpnp/backend_mathematical.pyx:318]: Maybe we will reuse `dpnp_cumsum` here?

File: dpnp/backend_mathematical.pyx [hunk 5]
```diff
@@ -283,6 +285,66 @@ cpdef dparray dpnp_multiply(dparray x1, x2):
         return call_fptr_2in_1out(DPNP_FN_MULTIPLY, x1, x2, x1.shape)
 
 
+cpdef dparray dpnp_nancumprod(dparray x1):
```
Comment (densmirn) [inline dpnp/backend_mathematical.pyx:288]: Maybe we will reuse `dpnp_cumprod` here?

File: tests/test_mathematical.py [hunk 6]
```diff
@@ -24,3 +24,61 @@ def test_diff(array):
...
+        numpy.testing.assert_array_equal(expected, result)
+
+
+    def test_nancumprod_dim2_nan(self):
+        a = numpy.array([[1, 2, numpy.nan], [3, -4, -5]])
+        ia = dpnp.array(a)
+
+        result = dpnp.nancumsum(ia)
+        expected = numpy.nancumsum(a)
+        numpy.testing.assert_array_equal(expected, result)
```
Comment (densmirn) [inline tests/test_mathematical.py:84]: Please insert blank line to the end of script.

**Review by densmirn — APPROVED** [review]

File: dpnp/backend_mathematical.pyx (hunk 4 above)
Comment (shssf) [reply dpnp/backend_mathematical.pyx:297]: with additional function parameter like `bool usenan`

File: dpnp/backend_mathematical.pyx [hunk 7]
```diff
@@ -120,20 +124,25 @@ cpdef dparray dpnp_cumprod(dparray x1):
         'float64': dpnp.float64
     }
 
-    res_type = types_map[x1.dtype.name]
+    res_type = types_map[x1_dtype.name]
```
Comment (densmirn) [inline dpnp/backend_mathematical.pyx:127]:
```suggestion
    res_type = types_map[x1.dtype.name]
```

File: dpnp/backend_mathematical.pyx [hunk 8]
```diff
@@ -142,14 +151,17 @@ cpdef dparray dpnp_cumsum(dparray x1):
...
```
Comment (densmirn) [inline dpnp/backend_mathematical.pyx:154]:
```suggestion
    res_type = types_map[x1.dtype.name]
```

File: dpnp/backend_mathematical.pyx [hunk 9]
```diff
@@ -58,6 +58,8 @@ __all__ += [
     "dpnp_minimum",
     "dpnp_modf",
     "dpnp_multiply",
+    "dpnp_nancumprod",
+    "dpnp_nancumsum",
```
Comment (densmirn) [inline dpnp/backend_mathematical.pyx:62]:
```suggestion
```

File: tests/test_mathematical.py [hunk 10]
```diff
@@ -24,3 +24,61 @@ def test_diff(array):
...
+
+
+class TestNanCumProd:
+
+    def test_nancumprod_dim1_int(self):
```
Comment (shssf) [inline tests/test_mathematical.py:60]: These 3 tests are the same but uses diffrent input data. Maybe it is better to use `pytest` features to combine tests into one with input variations. Example:
https://github.com/IntelPython/dpnp/blob/66b880c0483c3e803eca9edbbb6a07da27fc3020/tests/third_party/cupy/fft_tests/test_fft.py#L39-L62

**Pull request merged**